src/bin/clean.rs

```diff
@@ -0,0 +1,31 @@
+use std::fs;
+use std::path::PathBuf;
+use clap::{App, ArgMatches, SubCommand};
+use mdbook::MDBook;
+use mdbook::errors::*;
+use get_book_dir;
+
+// Create clap subcommand arguments
+pub fn make_subcommand<'a, 'b>() -> App<'a, 'b> {
+    SubCommand::with_name("clean")
+        .about("Delete built book")
+        .arg_from_usage(
+            "-d, --dest-dir=[dest-dir] 'The directory of built book{n}(Defaults to ./book when \
+             omitted)'",
+        )
+}
+
+// Clean command implementation
+pub fn execute(args: &ArgMatches) -> ::mdbook::errors::Result<()> {
+    let book_dir = get_book_dir(args);
+    let book = MDBook::load(&book_dir)?;
+
+    let dir_to_remove = if let Some(dest_dir) = args.value_of("dest-dir") {
+        PathBuf::from(dest_dir)
+    } else {
+        book.config.build.build_dir
```

Comment:
This is *almost* right. The problem is that `build_dir` is relative to a **book's root directory**, so when calling `mdbook clean` from a different directory (e.g. `mdbook clean /path/to/book`) it'll delete `build_dir` relative to the **current directory**.

This should really be `book.root.join(&book.config.build.build_dir)`, it will join the `book.root` path and `build_dir` to get the full path to the build directory.

Reply:
Did.